models/monitoring.ts

```diff
@@ -25,43 +24,49 @@ export interface IMonitoringWithMetaData extends IMonitoring {
   apiName: string;
 }
 
-const getMonitorings = async (): Promise<IMonitoring[]> => {
-  try {
-    return await clientMonitorings();
-  } catch (e: any) {
-    logErrorInSentry(e, {
-      errorName: 'Error while fetching monitoring',
-    });
-    return [];
-  }
-};
-
 export const getMonitorsByAdministration = async (): Promise<{
-  [key: string]: IMonitoringWithMetaData[];
+  [key: string]: (IMonitoringWithMetaData | IAPINotRespondingError)[];
 }> => {
-  const monitoringsFromUptimeRobot = await getMonitorings();
-
-  return Object.values(administrationsMetaData).reduce(
-    (allMonitorsByAdministration, { apiMonitors = [], administrationEnum }) => {
-      const administrationMonitors = apiMonitors.map(
-        ({ id, apigouvLink = null, apiName = null, apiSlug = null }) => {
-          const monitoring = monitoringsFromUptimeRobot.find(
-            (monitor) => monitor.id === id
-          );
+  const allMonitoringsByAdministration = {} as {
+    [key: string]: (IMonitoringWithMetaData | IAPINotRespondingError)[];
+  };
 
-          return {
-            apigouvLink,
-            apiSlug,
-            apiName,
-            ...monitoring,
-          };
-        }
+  for (let { apiMonitors = [], administrationEnum } of Object.values(
+    administrationsMetaData
+  )) {
+    const hasMonitors = apiMonitors.filter(({ id }) => !!id).length > 0;
+    if (hasMonitors) {
+      const monitorings = await Promise.all(
+        apiMonitors
+          .filter(({ id }) => !!id)
+          .map(
+            async ({
+              id,
+              apigouvLink = null,
+              apiName = null,
+              apiSlug = null,
+            }) => {
+              try {
+                const monitoring = await clientMonitoring(id);
+                return {
+                  apigouvLink,
+                  apiSlug,
+                  apiName,
+                  administrationEnum,
+                  ...monitoring,
+                } as IMonitoringWithMetaData;
+              } catch (e: any) {
+                logErrorInSentry(e, {
+                  errorName: 'Error while fetching monitoring',
+                });
+                throw e;
```

Comment:
Not sure about all this @johangirod : 

- not sure this is an actual promise all and not sequential calls
- not sure about the behaviour when it fails. Should we wrap it into `APINotResponding` ? As the page is cached and revalidated every 30 seconds, I believe it is better to let it fail. But as we do a LOT of call to their API I wonder if this is a good idea

Reply:
`await` works inside `for` loop but it will end up in sequential call, waiting for the current iteration to resolve before starting the next one. I think it would be clearer to construct the returned object doing the following : 

1. Get all API monitor ids (monitorIds) 
2. Construct an object `Array<Promise<id, IMonitor | IAPINotResponding >`
3. `Object.fromEntries(Promise.all()) de cet objet
4. Reconstruire l'objet de sortie à partir de cela